### PR TITLE
OP-10566: APIS to entended the session and retrieve session timeout.

### PR DESCRIPTION
**Pull request by rahul-chekuri**
These APIs are triggered from UI as part of session management. Below is the flow chart of session management.
Screenshot from 2021-12-24 16-47-26
.

### CODE REVIEW

**Comment by gopaljayanthi** [comment]
This PR  triggered this build in this execution, please merge to master only if this succeeds:  https://spin.prod.opsmx.com/#/applications/gcp-gitprtrigger/executions/details/01FQP6G3VDFF91NA56948D57BR

File: gate-web/src/main/groovy/com/opsmx/spinnaker/gate/controllers/OpsmxSessionController.groovy [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpSession
+
+@RequestMapping("/session")
+@RestController
+@Slf4j
+class OpsmxSessionController {
+
+  @Value('${server.session.timeout-in-seconds:3600}')
+  private int sessionTimeout
```
Comment (Pranav-b-7) [inline gate-web/src/main/groovy/com/opsmx/spinnaker/gate/controllers/OpsmxSessionController.groovy:38]: Remove the private modifier , I dont think groovy supports it.

File: gate-web/src/main/groovy/com/opsmx/spinnaker/gate/controllers/OpsmxSessionController.groovy [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  @ApiOperation(value = "get session timeout")
+  @GetMapping(value = "/getSessionTimeout")
+  Integer getSessionTimeout() {
+    log.info("sessionTimeout : {}" , sessionTimeout)
+    return sessionTimeout
+  }
+
+  @ApiOperation(value = "extend session by given duration")
+  @GetMapping(value = "/extendSession")
```
Comment (Pranav-b-7) [inline gate-web/src/main/groovy/com/opsmx/spinnaker/gate/controllers/OpsmxSessionController.groovy:48]: This is an update operation. So change it to @PutMapping

**Review by Pranav-b-7 — CHANGES_REQUESTED** [review]

**Comment by gopaljayanthi** [comment]
This PR  triggered this build in this execution, please merge to master only if this succeeds:  https://spin.prod.opsmx.com/#/applications/gcp-gitprtrigger/executions/details/01FQPC9K961HNRN3WCS0YP7V43

**Review by Pranav-b-7 — APPROVED** [review]

**Pull request merged**